drivers/mtd/mtd.c

```diff
@@ -139,24 +140,23 @@ int mtd_write(mtd_dev_t *mtd, const void *src, uint32_t addr, uint32_t count)
 
 #ifdef MODULE_MTD_WRITE_PAGE
 int mtd_write_page(mtd_dev_t *mtd, const void *data, uint32_t page,
-                   uint32_t offset, uint32_t len)
+                   uint32_t offset, uint32_t count)
 {
     if (!mtd || !mtd->driver) {
         return -ENODEV;
     }
 
     if (mtd->work_area == NULL) {
-        return mtd_write_page_raw(mtd, data, page, offset, len);
+        return mtd_write_page_raw(mtd, data, page, offset, count);
     }
 
     int res;
     uint8_t *work = mtd->work_area;
     const uint32_t sector = page / mtd->pages_per_sector;
-    const uint32_t sector_page = sector * mtd->pages_per_sector;
     const uint32_t sector_size = mtd->pages_per_sector * mtd->page_size;
 
     /* copy sector to RAM */
-    res = mtd_read_page(mtd, work, sector_page, 0, sector_size);
+    res = mtd_read_page(mtd, work, page, 0, sector_size);
```

Comment:
I don't think this is correct.
The idea of this function is to read the whole sector to memory, erase it on flash, modify it in memory and then write it back to flash.

For this to work we have to read at the *start* of the *sector*.

`page` could be anywhere in the sector.

Reply:
Ok possible I made a really stupid mistake, but: sector_page = sector * mtd->pages_per_sector = page / mtd->pages_per_sector * mtd->pages_per_sector = page. This is just a a simplification of the equation.

Reply:
This is integer arithmetic. `sector = page / mtd->pages_per_sector` will calculate which sector the page belongs to, e.g. if you have 

    page = 23;
    pages_per_sector = 8;

Then `sector` will be `23/8 = 2`. This is then used to calculate the start page of the sector: `2 * 8 = 16`.

Reply:
Ah there you go :P stupid mistake. Thanks ! Will revert this.